contrib/babelfishpg_tds/src/backend/tds/tdsrpc.c

```diff
@@ -801,6 +804,7 @@ SPExecute(TDSRequestSP req)
 	 */
 	if (pltsql_plugin_handler_ptr->stmt_needs_logging || TDS_DEBUG_ENABLED(TDS_DEBUG2))
 	{
+		MemoryContext curr = CurrentMemoryContext;
```

Comment:
it seems stmt needs logging block can be a utility function with appropriate parameter to put this code at one place.